defuse/src/contract/accounts/account/mod.rs

```diff
@@ -149,12 +149,12 @@ impl Account {
         self.nonces.commit(nonce)
     }
 
-    /// Clears the nonce if it was expired.
-    /// Returns whether the nonces was cleared. If the nonce has not expired yet, then returns `false`,
+    /// Clears the all nonces with corresponding prefix if it was expired/invalidated.
+    /// Returns whether the nonces was cleared,
     /// regardless of whether it was previously committed or not.
     #[inline]
-    pub fn clear_expired_nonce(&mut self, nonce: U256) -> bool {
-        self.nonces.clear_expired(nonce)
+    pub fn cleanup_nonce_by_prefix(&mut self, nonce: Nonce) -> bool {
```

Comment:
Be explicit about pefix:
```suggestion
    pub fn cleanup_nonce_by_prefix(&mut self, prefix: [u8; 31]) -> bool {
```